lib/puppet/provider/foreman_host/rest_v3.rb

```diff
@@ -0,0 +1,55 @@
+Puppet::Type.type(:foreman_host).provide(:rest_v3, :parent => Puppet::Type.type(:foreman_resource).provider(:rest_v3)) do
+  confine :feature => [:json, :oauth]
+
+  def exists?
+    !id.nil?
+  end
+
+  def create
+    path = "api/v2/hosts/facts"
+    payload = {
+      :name => resource[:name],
+      :certname => resource[:name],
+      :facts => resource[:facts]
+    }
+    r = request(:post, path, {}, payload.to_json)
```

Comment:
Personal nit, could this at least be `req` or `request`. I find single letter names mental overhead when reading code.